src/unstract/sdk/index.py

```diff
@@ -138,58 +152,68 @@ def index_file(
         """
         doc_id = self.generate_file_id(
             tool_id=tool_id,
-            vector_db=vector_db,
-            embedding=embedding_type,
-            x2text=x2text_adapter,
+            vector_db=vector_db_instance_id,
+            embedding=embedding_instance_id,
+            x2text=x2text_instance_id,
             chunk_size=str(chunk_size),
             chunk_overlap=str(chunk_overlap),
             file_path=file_path,
             file_hash=file_hash,
         )
         self.tool.stream_log(f"Checking if doc_id {doc_id} exists")
 
-        # Get embedding instance
-        embedd_helper = ToolEmbedding(tool=self.tool)
-        embedding_li = embedd_helper.get_embedding(adapter_instance_id=embedding_type)
-        embedding_dimension = embedd_helper.get_embedding_length(embedding_li)
+        try:
+            embedding = Embedding(
+                tool=self.tool,
+                adapter_instance_id=embedding_instance_id,
+                usage_kwargs=usage_kwargs,
+            )
+        except SdkError as e:
+            self.tool.stream_log(
+                f"Error loading {embedding_instance_id}", level=LogLevel.ERROR
+            )
+            raise SdkError(f"Error loading {embedding_instance_id}: {e}")
 
-        # Get vectorDB instance
-        vdb_helper = ToolVectorDB(
-            tool=self.tool,
-        )
-        vector_db_li = vdb_helper.get_vector_db(
-            adapter_instance_id=vector_db,
-            embedding_dimension=embedding_dimension,
-        )
+        try:
+            vector_db = VectorDB(
+                tool=self.tool,
+                adapter_instance_id=vector_db_instance_id,
+                embedding=embedding,
+            )
+        except SdkError as e:
+            self.tool.stream_log(
+                f"Error loading {vector_db_instance_id}", level=LogLevel.ERROR
+            )
+            raise SdkError(f"Error loading {vector_db_instance_id}: {e}")
 
         # Checking if document is already indexed against doc_id
         doc_id_eq_filter = MetadataFilter.from_dict(
             {"key": "doc_id", "operator": FilterOperator.EQ, "value": doc_id}
         )
         filters = MetadataFilters(filters=[doc_id_eq_filter])
         q = VectorStoreQuery(
-            query_embedding=embedding_li.get_query_embedding(" "),
+            query_embedding=embedding.get_query_embedding(" "),
             doc_ids=[doc_id],
             filters=filters,
         )
 
         doc_id_found = False
         try:
-            n: VectorStoreQueryResult = vector_db_li.query(query=q)
+            n: VectorStoreQueryResult = vector_db.query(query=q)
```

Comment:
Dont we require to close this connection on excepetion?

Reply:
Yes. Thank you for reminding :) The close() was just implemented only now for all vector DBs. Still not in main branch. Can I do it in phase 2?  I need to incorporate it in all places and then test it.